From 1db0f8a187f2a4213c08e56d7146aa472de6b440 Mon Sep 17 00:00:00 2001
From: Divya Prakash <divya1.prakash@intel.com>
Date: Tue, 13 Jan 2015 14:25:39 +0530
Subject: [PATCH 13/15] ASoC: Intel: Enable micbias1 for CHT FFD mic detection

Issue: GMINL-6292

This patch is to enable micbias1 for CHT FFD mic detection

Change-Id: Ia7fa109de83262e23d04b7bebb603d4423360a32
Signed-off-by: Divya Prakash <divya1.prakash@intel.com>
Signed-off-by: Fabien Marotte <fabien.marotte@intel.com>
---
 sound/soc/intel/board/cht_bl_dpcm_rt5672.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
index 1a34447..4fc982e 100644
--- a/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
+++ b/sound/soc/intel/board/cht_bl_dpcm_rt5672.c
@@ -30,6 +30,7 @@
 #include <linux/slab.h>
 #include <linux/vlv2_plat_clock.h>
 #include <linux/mutex.h>
+#include <linux/dmi.h>
 #include <asm/platform_cht_audio.h>
 #include <asm/intel-mid.h>
 #include <sound/pcm.h>
@@ -94,9 +95,15 @@ static inline void cht_force_enable_pin(struct snd_soc_codec *codec,
 static inline void cht_set_codec_power(struct snd_soc_codec *codec,
 								int jack_type)
 {
+	const char *board_name;
 	switch (jack_type) {
 	case SND_JACK_HEADSET:
-		cht_force_enable_pin(codec, "micbias2", true);
+		board_name = dmi_get_system_info(DMI_BOARD_NAME);
+		pr_debug("Setting the micbias for %s\n", board_name);
+		if (strcmp(board_name, "Cherry Trail FFD") == 0)
+			cht_force_enable_pin(codec, "micbias1", true);
+		else
+			cht_force_enable_pin(codec, "micbias2", true);
 		cht_force_enable_pin(codec, "JD Power", true);
 		cht_force_enable_pin(codec, "Mic Det Power", true);
 		break;
-- 
1.9.1

